### PR TITLE
feat(pingcap/tiflash): add refactored jobs for release-8.1

### LINKED ISSUE #3063: [TiFlash@release-8.1] Support more integration pipeline in VerfiyCI of tidb lts branches

### PR DESCRIPTION
**Pull request by wuhuizuo**
Fixes #3063

Signed-off-by: wuhuizuo <wuhuizuo@126.com>

### CODE REVIEW

File: pipelines/pingcap/tiflash/release-8.1/pod-pull_unit-test.yaml [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+          cpu: "500m"
+          memory: "500Mi"
+        limits:
+          cpu: "500m"
+          memory: "500Mi"
+      volumeMounts:
+      - mountPath: "/home/jenkins/agent/rust"
+        name: "volume-0"
+        readOnly: false
+      - mountPath: "/home/jenkins/agent/ccache"
```
Comment (wuhuizuo) [inline pipelines/pingcap/tiflash/release-8.1/pod-pull_unit-test.yaml:52]: @purelind we should remove this container. It just a commander container which Jenkins will append it automatically.

File: prow-jobs/pingcap/tiflash/release-8.1-presubmits.yaml [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+  branches: &branches
+    - ^release-8\.1(\.\d+)?(-\d+)?(-v[\.\d]+)?$
+  skip_if_only_changed: &skip_if_only_changed
+    "(\\.(md|png|jpeg|jpg|gif|svg|pdf)|Dockerfile|OWNERS|OWNERS_ALIASES)$"
+presubmits:
+  pingcap/tiflash:
+    - name: pingcap/tiflash/release-8.1/pull_unit_test
+      agent: jenkins
+      decorate: false # need add this.
+      skip_if_only_changed: *skip_if_only_changed
```
Comment (purelind) [inline prow-jobs/pingcap/tiflash/release-8.1-presubmits.yaml:12]:
```suggestion
      # skip_if_only_changed: *skip_if_only_changed
      always_run: false
```

File: prow-jobs/pingcap/tiflash/release-8.1-presubmits.yaml [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+      context: pull-unit-test
+      skip_report: true
+      optional: true
+      trigger: "(?m)^/test (?:.*? )?(pull-unit-test)(?: .*?)?$"
+      rerun_command: "/test pull-unit-test"
+      branches: *branches
+    - name: pingcap/tiflash/release-8.1/pull_integration_test
+      agent: jenkins
+      decorate: false # need add this.
+      skip_if_only_changed: *skip_if_only_changed
```
Comment (purelind) [inline prow-jobs/pingcap/tiflash/release-8.1-presubmits.yaml:22]:
```suggestion
      # skip_if_only_changed: *skip_if_only_changed
      always_run: false
```

File: prow-jobs/pingcap/tiflash/release-8.1-presubmits.yaml [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+      context: pull-integration-test
```
Comment (purelind) [inline prow-jobs/pingcap/tiflash/release-8.1-presubmits.yaml:23]:
```suggestion
      context: wip/pull-integration-test
```

File: prow-jobs/pingcap/tiflash/release-8.1-presubmits.yaml [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+    - ^release-8\.1(\.\d+)?(-\d+)?(-v[\.\d]+)?$
+  skip_if_only_changed: &skip_if_only_changed
+    "(\\.(md|png|jpeg|jpg|gif|svg|pdf)|Dockerfile|OWNERS|OWNERS_ALIASES)$"
+presubmits:
+  pingcap/tiflash:
+    - name: pingcap/tiflash/release-8.1/pull_unit_test
+      agent: jenkins
+      decorate: false # need add this.
+      skip_if_only_changed: *skip_if_only_changed
+      context: pull-unit-test
```
Comment (purelind) [inline prow-jobs/pingcap/tiflash/release-8.1-presubmits.yaml:13]:
```suggestion
      context: wip/pull-unit-test
```

File: pipelines/pingcap/tiflash/release-8.1/pod-pull_unit-test.yaml (hunk 1 above)
Comment (purelind) [reply pipelines/pingcap/tiflash/release-8.1/pod-pull_unit-test.yaml:52]: There is a cache presence check in the pipeline that will definitely be executed in the jnlp container, so it cannot be removed for now.

**Review by purelind — APPROVED** [review]
/lgtm

**Pull request merged**